File: backend/app/schemas/payment.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class PaymentProcessRequest(BaseModel):
    booking_id: int = Field(gt=0)
    method: str = Field(pattern="^(WALLET|CARD|CASH)$")


class PaymentResponse(BaseModel):
    id: int
    user_id: int
    booking_id: int
    transaction_id: Optional[int]
    amount: float
    method: str
    status: str
    validated_by: Optional[int]
    validated_at: Optional[datetime]
    created_at: datetime

    model_config = {"from_attributes": True}


class PaymentValidateResponse(BaseModel):
    id: int
    status: str
    validated_by: int
    validated_at: datetime
    message: str = "Payment validated successfully"

File: flutter_app/lib/features/payment/domain/entities/payment.dart
import 'package:equatable/equatable.dart';

class Payment extends Equatable {
  final int id;
  final int userId;
  final int bookingId;
  final int? transactionId;
  final double amount;
  final String method;
  final String status;
  final int? validatedBy;
  final DateTime? validatedAt;
  final DateTime createdAt;

  const Payment({
    required this.id,
    required this.userId,
    required this.bookingId,
    this.transactionId,
    required this.amount,
    required this.method,
    required this.status,
    this.validatedBy,
    this.validatedAt,
    required this.createdAt,
  });

  bool get isCompleted => status == 'COMPLETED';
  bool get isPending => status == 'PENDING';

  @override
  List<Object?> get props => [id, bookingId, amount, method, status];
}
